### DISCUSSION TITLE
Best way to minimize RAM and CPU usage?

### DISCUSSION THREAD

**Discussion by throwaccount11** [savonet/liquidsoap#1767] [Q&A]
I'm going to be running multiple stations on a single machine, so I'd need to have minimal CPU/RAM usage. I'm not sure if this would work, but is there a way to disable decoding and encoding so that it would output opus audio from opus files?

Thanks.

**Comment by smimram** [comment]
This is not currently possible, however @toots is working intensively on this and it should be doable in Liquidsoap 2.0.

**Comment by throwaccount11** [comment]
Thanks, do you know when this might be finished/released? Also do you know of any other ways to reduce CPU/RAM usage?

**Comment by toots** [comment]
Hi! Yes, if your usage does not require modifying the data itself, you should be able to use our new support for encoded data. If you provide more details, I'd be happy to explain how to.

**Reply by rastawarez** [reply]
I suppose fade , crossfade , cue_cut  all of them are data modificators or not ? It is decoded to raw , processed and encoded to required format

**Reply by smimram** [reply]
Yes, fading or cutting tracks requires decoding audio.

**Comment by Charbacca** [comment]
Hi, sorry for this "noob"-question but I am using Webcaster Client (https://webcast.github.io/webcaster/) to stream to my Liquidsoap-Server and provide it to Icecast2. As far as I understand it, the audio-data comes already as MP3-format from the Webcaster-Client, so there is actually no encoding necessary by Liquidsoap, isn't it?
At the moment I am using Liquidsoap 1.4.3 and see for every stream I open a CPU-usage of about 4-5%. This is a problem for me, because I would like to use more then 20 different streams on my server simultaneously.

Can I run Liquidsoap already without encoding to reduce CPU usage on the server?

**Comment by smimram** [comment]
Yes Liquidsoap 2.0 has support for manipulating streams without re-encoding, using encoders of the form `%ffmpeg(%audio.copy)` (see [the doc](https://www.liquidsoap.info/doc-dev/encoding_formats.html) or [the book](https://www.liquidsoap.info/doc-dev/book.html)).
